docs/user-guide/ingress.md

```diff
@@ -0,0 +1,240 @@
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Ingress
+
+#### Terminology
+
+Throughout this doc you will see a few terms that are sometimes used interchangably elsewhere, that might cause confusion. This section attempts to clarify them.
+
+* Node: A single virtual or physical machine in a Kubernetes cluster.
+* Cluster: A group of nodes firewalled from the internet, that are the primary compute resources managed by Kubernetes.
+* Edge router: A router that enforces the firewall policy for your cluster. This could be a gateway managed by a cloudprovider or a physical piece of hardware.
+* Cluster network: A set of links, logical or physical, that facilitate communication within a cluster according to the [Kubernetes networking model](https://github.com/kubernetes/kubernetes/blob/release-1.0/docs/admin/networking.md). Examples of a Cluster network include Overlays such as [flannel](https://github.com/coreos/flannel#flannel) or SDNs such as [OVS](https://github.com/kubernetes/kubernetes/blob/release-1.0/docs/admin/ovs-networking.md).
+* Service: A Kubernetes [Service](https://github.com/kubernetes/kubernetes/blob/release-1.0/docs/user-guide/services.md) that identifies a set of pods using label selectors. Unless mentioned otherwise, Services are assumed to have virtual IPs only routable within the cluster network.
+
+## What is Ingress?
+
+Typically, services and pods have IPs only routable by the cluster network. All traffic that ends up at an edge router is either dropped or forwarded elsewhere. Conceptually, this might look like:
+
+```
+    internet
+        |
+  ------------
+  [ Services ]
+```
+
+An Ingress is a collection of rules that allow inbound connections to reach the cluster services.
+
+```
+    internet
+        |
+   [ Ingress ]
+   --|-----|--
+   [ Services ]
+```
+
+It can be configured to give services externally-reachable urls, load balance traffic, terminate SSL, offer name based virtual hosting etc. Users request ingress by POSTing the Ingress resource to the API server. An [Ingress controller](#ingress-controllers) is responsible for fulfilling the Ingress, usually with a loadbalancer, though it may also configure your edge router or additional frontends to help handle the traffic in an HA manner.
+
+## The Ingress Resource
+
+A minimal Ingress might look like:
+
+```yaml
+01. apiVersion: extensions/v1beta1
+02. kind: Ingress
+03. metadata:
+04.  name: hostlessendpoint
+05. spec:
+06.  rules:
+07.  - http:
+08.      paths:
+09.      - path: /hostless
+10.        backend:
+11.          serviceName: test
+12.          servicePort: 80
+```
+
+[Download example](ingress.yaml?raw=true)
+
+*POSTing this to the API server will have no effect if you have not configured an [Ingress controller](#ingress-controllers).*
+
+__Lines 1-4__: Resource metadata. An Ingress controller might use this to tag shared resources such as loadbalancers with a label or annotation.
```

Comment:
the text I've used in other docs for this is:
```
As with all other Kubernetes config, an Ingress needs `apiVersion`, `kind`, and `metadata` fields.  For
general information about working with config files, see [here](simple-yaml.md),
[here](configuring-containers.md), and [here](working-with-resources.md).
```

The current remark about labels an annotations is not relevant because your example doesn't use them, and here isn't the place to do that.